src/ops/delay.c

```diff
@@ -4,35 +4,56 @@
 #include "teletype.h"
 #include "teletype_io.h"
 
+static void delay_common_add(scene_state_t *ss, exec_state_t *es, 
+                        int16_t i, int16_t delay_time,
+                        const tele_command_t *post_command);
+
 static void mod_DEL_func(scene_state_t *ss, exec_state_t *es,
                          command_state_t *cs,
                          const tele_command_t *post_command);
 
 static void op_DEL_CLR_get(const void *data, scene_state_t *ss,
                            exec_state_t *es, command_state_t *cs);
 
+static void mod_DEL_X_func(scene_state_t *ss, exec_state_t *es,
+                         command_state_t *cs,
+                         const tele_command_t *post_command);
+
+static void mod_DEL_R_func(scene_state_t *ss, exec_state_t *es,
+                         command_state_t *cs,
+                         const tele_command_t *post_command);
 
 const tele_mod_t mod_DEL = MAKE_MOD(DEL, mod_DEL_func, 1);
 const tele_op_t op_DEL_CLR = MAKE_GET_OP(DEL.CLR, op_DEL_CLR_get, 0, false);
+const tele_mod_t mod_DEL_X = MAKE_MOD(DEL.X, mod_DEL_X_func, 2);
+const tele_mod_t mod_DEL_R = MAKE_MOD(DEL.R, mod_DEL_R_func, 2);
+
+//common code to queue a delay shared between all delay ops
+//NOTE it is the responsibility of the callee to call tele_has_delays
+static void delay_common_add(scene_state_t *ss, exec_state_t *es, 
+                        int16_t i, int16_t delay_time,
+                        const tele_command_t *post_command) {
+        ss->delay.count++;
```

Comment:
i would add a check here to make sure it doesn't exceed `DELAY_SIZE` - imho it's a good idea to have safety checks in helper functions.

Reply:
ah good catch I can add that in.  on the same topic do you think it would be good to move ```tele_has_delays``` into the helper function as well?  the downside is it would get excess times times by ```DEL.X```, but would make the helper function slightly more robust for other future uses.

Reply:
i'd leave it out - the function has a well defined and logical contract (it adds a delay), `tele_has_delays` is outside of that (and only has meaning to where this function is called from).